lib/yoti/activity_details.rb

```diff
@@ -1,69 +1,167 @@
+    #
+    # @return [String]
+    #
     attr_reader :parent_remember_me_id
 
-    # @return [Hash] the decoded profile attributes
+    #
+    # The decoded profile attributes
+    #
+    # @deprecated replaced by :profile
+    #
+    # @return [Hash]
+    #
     attr_reader :user_profile
 
-    # @return [String] the selfie in base64 format
+    #
+    # Base64 encoded selfie image
+    #
+    # @return [String]
+    #
     attr_reader :base64_selfie_uri
 
-    # @return [Boolean] the age under/over attribute
+    #
+    # The age under/over attribute
+    #
+    # @return [Boolean]
+    #
     attr_reader :age_verified
 
+    #
+    # Receipt ID identifying a completed activity
+    #
+    # @return [String]
+    #
+    attr_reader :receipt_id
+
+    #
+    # Time and date of the sharing activity
+    #
+    # @return [Time]
+    #
+    attr_reader :timestamp
+
+    #
     # @param receipt [Hash] the receipt from the API request
     # @param decrypted_profile [Object] Protobuf AttributeList decrypted object containing the profile attributes
-    def initialize(receipt, decrypted_profile = nil)
+    #
+    def initialize(receipt, decrypted_profile = nil, decrypted_application_profile = nil)
       @remember_me_id = receipt['remember_me_id']
       @user_id = @remember_me_id
+      @receipt_id = receipt['receipt_id']
       @parent_remember_me_id = receipt['parent_remember_me_id']
       @outcome = receipt['sharing_outcome']
-      @user_profile = {}
-      @extended_profile = {}
-      process_decrypted_profile(decrypted_profile)
+      @timestamp = receipt['timestamp'] ? Time.parse(receipt['timestamp']) : nil
+      @extended_user_profile = process_decrypted_profile(decrypted_profile)
+      @extended_application_profile = process_decrypted_profile(decrypted_application_profile)
+      @user_profile = @extended_user_profile.map do |name, attribute|
+        [name, attribute.value]
+      end.to_h
     end
 
-    # @return [Hash] a JSON of the address
+    #
+    # The user's structured postal address as JSON
+    #
+    # @deprecated replaced by Profile.structured_postal_address
+    #
+    # @return [Hash]
+    #
     def structured_postal_address
-      @user_profile['structured_postal_address']
+      user_profile['structured_postal_address']
     end
 
-    # @return [Profile] of Yoti user
+    #
+    # The user profile with shared attributes and anchor information, returned
+    # by Yoti if the request was successful
+    #
+    # @return [Profile]
+    #
     def profile
-      Yoti::Profile.new(@extended_profile)
+      Yoti::Profile.new(@extended_user_profile)
+    end
+
+    #
+    # Profile of an application, with convenience methods to access well-known attributes
+    #
+    # @return [ApplicationProfile]
+    #
+    def application_profile
+      Yoti::ApplicationProfile.new(@extended_application_profile)
     end
 
     protected
 
+    #
+    # Process the decrypted profile to into hash
```

Comment:
`Process the decrypted profile into a hash`?

Reply:
It processes the profile into https://docs.ruby-lang.org/en/2.0.0/Hash.html - should we reword this?

Reply:
```suggestion
    # Process the decrypted profile to into key-value hash
```

Reply:
👍